### PR TITLE
Set the DNS operator up for Lighthouse

### PR DESCRIPTION
**Pull request by skitt**
Signed-off-by: Stephen Kitt <skitt@redhat.com>

### CODE REVIEW

File: pkg/subctl/lighthouse/dns/ensure.go [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+	"k8s.io/apimachinery/pkg/util/wait"
+	clientset "k8s.io/client-go/kubernetes"
+	clientsetifc "k8s.io/client-go/kubernetes/typed/apps/v1"
+	"k8s.io/client-go/rest"
+
+	"github.com/submariner-io/submariner-operator/pkg/internal/cli"
+)
+
+const (
+	operatorImage           = "quay.io/aswinsuryan/lighthouse-cluster-dns-operator:v0.0.1"
```
Comment (mangelajo) [inline pkg/subctl/lighthouse/dns/ensure.go]: Did we also need to modify the cluster-dns-operator? :)

File: pkg/subctl/lighthouse/dns/ensure.go [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+)
+
+func Ensure(status *cli.Status, config *rest.Config) error {
+	clientSet, err := clientset.NewForConfig(config)
+	if err != nil {
+		return err
+	}
+
+	// Disable ClusterVersionOperator
+	_, err = scaleDeployment(clientSet.AppsV1().Deployments("openshift-cluster-version"), "cluster-version-operator", 0)
```
Comment (mangelajo) [inline pkg/subctl/lighthouse/dns/ensure.go]: Isn't this very openshift specific?, what about others? Can we add comments about the reasoning for this?

File: pkg/subctl/lighthouse/dns/ensure.go [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+	if err != nil {
+		return err
+	}
+	status.QueueSuccessMessage("Disabled the cluster version operator")
+
+	// Fix up cluster role
+	clusterRole, err := clientSet.RbacV1().ClusterRoles().Get("openshift-dns-operator", metav1.GetOptions{})
```
Comment (mangelajo) [inline pkg/subctl/lighthouse/dns/ensure.go]: Same here. We need to think about how to do this in other cases, not only satisfy openshift, although I understand probably for a POC is what we have, we cannot consider this feature finished or ready until we handle something that would work community wide.

Comment (skitt) [reply pkg/subctl/lighthouse/dns/ensure.go]: Yes, this whole patch is OpenShift-specific and very hacky. We shouldn’t have to deal with any of this stuff, and the cluster ends up somewhat broken as a result...

**Comment by skitt** [comment]
This should now be transparent on non-OpenShift deployments (I’ll check tomorrow).

**Pull request merged**